src/common/headers.rs

```diff
@@ -289,6 +295,29 @@ impl Headers {
     pub fn set_accept(&mut self, media_type: MediaType) {
         self.accept = media_type;
     }
+
+    /// Insert a new custom header and value pair into the `HashMap`.
+    pub fn insert_custom_header_line(
+        &mut self,
+        key: String,
+        value: String,
+    ) -> Result<(), RequestError> {
+        if self.custom_map.len() < MAX_CUSTOM_HEADER_LINES {
+            self.custom_map.insert(key, value);
+            return Ok(());
+        }
+        Err(RequestError::CustomHeadersLimitExceeded)
+    }
+
+    /// Returns a custom header value.
+    pub fn get_custom_header_value(&self, key: &str) -> Option<&String> {
```

Comment:
nit:
```suggestion
    pub fn custom_header_value(&self, key: &str) -> Option<&String> {
```

Reply:
I removed `get_custom_header_value()` instead, based on the conversation above.